### PR TITLE
- add batch handler for moab

### PR DESCRIPTION
**Pull request by chanwilson**
This is a minor adaptation of the pbs handler for MOAB use.  In active use at the GFDL.

### CODE REVIEW

File: lib/cylc/batch_sys_handlers/moab.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+class MoabHandler(object):
+
+    "Moab batch system job submission and manipulation."
+
+    DIRECTIVE_PREFIX = "#PBS "
+    KILL_CMD_TMPL = "mjobctl -c '%(job_id)s'"
+    # N.B. The "qstat JOB_ID" command returns 1 if JOB_ID is no longer in the
+    # system, so there is no need to filter its output.
+#TODO: how is this output used?
```
Comment (hjoliver) [inline lib/cylc/batch_sys_handlers/moab.py]: Do you mean the qstat output referred to in the preceding comment?

File: lib/cylc/batch_sys_handlers/moab.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    REC_ID_FROM_SUBMIT_OUT = re.compile(r"""\A\s*(?P<id>\S+)\s*\Z""")
+    SUBMIT_CMD_TMPL = "msub '%s'"
+
+    def format_directives(self, job_conf):
+        """Format the job directives for a job file."""
+        job_file_path = job_conf["job file path"].replace(r"$HOME/", "")
+        directives = job_conf["directives"].__class__()  # an ordereddict
+
+        # Old versions of PBS (< 11) requires jobs names <= 15 characters.
+        # Version 12 appears to truncate the job name to 15 characters if it is
```
Comment (hjoliver) [inline lib/cylc/batch_sys_handlers/moab.py]: I don't know how moab relates to PBS (I see that DIRECTIVE_PREFIX remains "#PBS"?) ... but could you just remove this comment if it is not relevant to moab?

File: lib/cylc/batch_sys_handlers/moab.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        # longer.
+        directives["-N"] = (
+            job_conf["task id"] + "." + job_conf["suite name"])
+
+        #directives["-j"] = "oe"
```
Comment (hjoliver) [inline lib/cylc/batch_sys_handlers/moab.py]: Can this (from pbs.py) be deleted?

**Comment by hjoliver** [comment]
@chanwilson - thanks for this; just a couple of comments above.  For future reference, to avoid having your master branch diverge from cylc's you should make changes (and then pull requests) on a new branch off of master in your clone.  To avoid shutting down this pull request you can leave it as is for now, but after this is merged you may need to delete your master branch and recreate it from the central cylc master.

File: lib/cylc/batch_sys_handlers/moab.py (hunk 1 above)
Comment (chanwilson) [inline lib/cylc/batch_sys_handlers/moab.py]: Yes, I'll remove that comment.

File: lib/cylc/batch_sys_handlers/moab.py (hunk 2 above)
Comment (chanwilson) [inline lib/cylc/batch_sys_handlers/moab.py]: Moab understand #PBS directives, so that part is accurate, but the commentary isn't.

**Comment by chanwilson** [comment]
@hjoliver - thanks for the feedback, I'll clean up the code as commented on.

**Comment by hjoliver** [comment]
@chanwilson - if you can get the aforementioned changes in, this can go in the next release...

**Comment by hjoliver** [comment]
(closing -superseded by #1533 - as @chanwilson seems to have gone AWOL :grimacing:)

**Pull request closed without merging**